### DISCUSSION TITLE
Tools compatible with Calibre 7?

### DISCUSSION THREAD

**Discussion by Two-For-Mirth** [noDRM/DeDRM_tools#539] [Q&A]
Since the latest stable update only mentions Calibre 6, I haven't upgraded to Calibre 7 for fear of loosing the tools. Today for the first time I got a worrying "Error Communicating with Device" when I plugged in my Kobo. I wanted to try to update Calibre to the latest version to see if that helps, but I don't want to do it if it means loosing the plug-in functionality. Thanks for any advice you can provide! Here's the error, by the way. 

calibre, version 6.29.0
ERROR: Error: Error communicating with device

CorruptError: database disk image is malformed

Traceback (most recent call last):
  File "calibre/gui2/device.py", line 104, in run
  File "calibre/gui2/device.py", line 558, in _books
  File "calibre/devices/kobo/driver.py", line 2028, in books
  File "src/cursor.c", line 240, in resetcursor
apsw.CorruptError: CorruptError: database disk image is malformed

**Comment by ElleKayEm** [comment]
I've been using both DeDRM and Obok 10.0.9 with calibre 7.  The only issue I had (adding a Kindle for Android key) was fixed in the alpha release of DeDRM.

I suggest asking at the MobileRead forums about that error with your Kobo.
